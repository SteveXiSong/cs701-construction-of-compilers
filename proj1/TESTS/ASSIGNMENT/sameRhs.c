#include <stdio.h>

int main() {
  int g1, g2;
  int a, b, c, w, x, y, z;
  scanf("%d", &g1);
  g2 = g1;
  a = g2 + 33;
  scanf("%d", &x);  /* other instruction (in same basic block) */
  b = g2 + 33;      /* redundant */
  scanf("%d", &y);  /* other instruction (in same basic block) */
  c = g2 + 33;      /* redundant */
  scanf("%d", &z);  /* other instruction (in same basic block) */
  g1 = a+b+c;
  printf("g1: %d\ng2: %d\na: %d\nb: %d\nc: %d\n", g1, g2, a, b, c);
}
