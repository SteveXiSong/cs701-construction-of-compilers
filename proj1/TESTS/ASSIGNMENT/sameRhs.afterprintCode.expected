
FUNCTION main

BASIC BLOCK entry
%1:	alloca	 XXX
%2:	alloca	 XXX
%3:	alloca	 XXX
%4:	alloca	 XXX
%5:	alloca	 XXX
%6:	alloca	 XXX
%7:	alloca	 XXX
%8:	alloca	 XXX
%9:	alloca	 XXX
%10:	call	 XXX %1 __isoc99_scanf
%11:	load	 %1
%12:	store	 %11 %2
%13:	load	 %2
%14:	add	 %13 XXX
%15:	store	 %14 %3
%16:	call	 XXX %7 __isoc99_scanf
%17:	load	 %2
%18:	add	 %17 XXX
%19:	store	 %18 %4
%20:	call	 XXX %8 __isoc99_scanf
%21:	load	 %2
%22:	add	 %21 XXX
%23:	store	 %22 %5
%24:	call	 XXX %9 __isoc99_scanf
%25:	load	 %3
%26:	load	 %4
%27:	add	 %25 %26
%28:	load	 %5
%29:	add	 %27 %28
%30:	store	 %29 %1
%31:	load	 %1
%32:	load	 %2
%33:	load	 %3
%34:	load	 %4
%35:	load	 %5
%36:	call	 XXX %31 %32 %33 %34 %35 printf
%37:	ret	 XXX
